packages/opentelemetry-instrumentation-anthropic/opentelemetry/instrumentation/anthropic/streaming.py

```diff
@@ -0,0 +1,143 @@
+import logging
+from opentelemetry.instrumentation.anthropic.utils import _set_span_attribute, should_send_prompts
+from opentelemetry.trace.status import Status, StatusCode
+from opentelemetry.semconv.ai import SpanAttributes
+
+from opentelemetry.instrumentation.anthropic.config import Config
+
+logger = logging.getLogger(__name__)
+
+
+def process_response_item(item, complete_response):
+    if item.type == 'message_start':
+        complete_response["model"] = item.message.model
+        complete_response["usage"] = item.message.usage
+    elif item.type == 'content_block_start':
+        index = item.index
+        if len(complete_response.get("events")) <= index:
+            complete_response["events"].append({"index": index, "text": ""})
+    elif item.type == 'content_block_delta' and item.delta.type == 'text_delta':
+        index = item.index
+        complete_response.get("events")[index]["text"] += item.delta.text
+    elif item.type == 'message_delta':
+        for event in complete_response.get("events", []):
+            event["finish_reason"] = item.delta.stop_reason
+
+
+def _set_token_usage(span, complete_response, prompt_tokens, completion_tokens):
+    total_tokens = prompt_tokens + completion_tokens
+    _set_span_attribute(span, SpanAttributes.LLM_USAGE_PROMPT_TOKENS, prompt_tokens)
+    _set_span_attribute(
+        span, SpanAttributes.LLM_USAGE_COMPLETION_TOKENS, completion_tokens
+    )
+    _set_span_attribute(span, SpanAttributes.LLM_USAGE_TOTAL_TOKENS, total_tokens)
+
+    _set_span_attribute(span, SpanAttributes.LLM_RESPONSE_MODEL, complete_response.get("model"))
+
+
+def _set_completions(span, choices):
```

Comment:
if this is a copy paste from the `__init__` - better to just re-use it from there (or use this method there). I want to avoid duplications as much as possible cause these methods change often and people tend to forget to look for all the places where a certain attribute is set.